src/msg/async/EventKqueue.cc

```diff
@@ -199,13 +199,17 @@ int KqueueDriver::resize_events(int newsize)
   ldout(cct,30) << __func__ << " kqfd = " << kqfd << "newsize = " << newsize 
                 << dendl;
   if(newsize > sav_max) {
-    sav_events = (struct SaveEvent*)realloc( sav_events, 
-                    sizeof(struct SaveEvent)*newsize);
-    if (!sav_events) {
+    void *_realloc = NULL;
+    _realloc = realloc(sav_events,
+                          sizeof(struct SaveEvent)*newsize);
+    if (_realloc == NULL) {
```

Comment:
@tchaikov realloc(ptr, size) will return NULL if it failed to allocate storage for the new size, and thus the block pointed by ptr was still not freed. And it will lead to possible memory leak. If we only add an assert(), the block for the previous sav_events is still not freed, even if sav_events is set to NULL upon a failed realloc() call.